### ISSUE TITLE
Unit.to_string contains whitespace for vounit format

### ISSUE THREAD

**Issue by juliantaylor** [astropy/astropy#4381]
according to the documentation the "vounit" format should be the VO 1.0 format but when using this in the `to_string` method one does not get a compliant VO representation of the unit which uses dot as a product and not whitespace:

```
import astropy.units
vo = astropy.units.format.VOUnit()
u = vo.parse("m s") # technically not VO syntax but leniency on input is ok
print u.to_string("vounit")
```

--> `m s`
This is not correct it should be `m.s`. The VO reference unity library does not accept this as VO format.
Its probably not a big deal as most unit standards accept space as product, but it is usually wise to be strict on what one produces even if one is lenient in what on accepts.
using astropy 1.1rc2

**Comment by embray** [comment]
I feel like there may be some history to this, but I need to double check.

**Comment by eteq** [comment]
I took a quick shot at implementing a fix to this in [my vounit-space-to-dot branch](https://github.com/eteq/astropy/tree/vounit-space-to-dot).  I think it works the way @juliantaylor wants, but a bunch of the tests are failing because they assume there is supposed to be a space.  

More troubling is the question of the "round-trip" test.  That is, there's a test that seems to assume reading in a string and then outputting it should round-trip, but of course that doesn't work if you want to accept e.g. "m s" but then re-process it into "m.s" for the vounit standard.  I suspect it's just a matter of altering the test rather than an important assumption being tested, but I'm not 100% sure... At any rate, for now I'll leave this be, but anyone who wants to pick up where my branch is right now is welcome to do so (@juliantaylor perhaps?)

**Comment by embray** [comment]
Yeah, when you instantiate a unit from a string, like `u.Unit('m.s')` or `u.Unit('m s')` the first parser it goes through by default is the "Generic" parser, which is very lenient (a product of units can be represented with a dot, a start, or a space in this format).  So when you pass in `u.Unit('m.s')` it gets parsed correctly, but it isn't explicitly *recognized* as a unit in vounit format.  So that's why in general it might not round-trip.

But I agree then when you *explicitly* output a unit with `to_string('vounit')` I don't know why it shouldn't do the right thing.

**Comment by embray** [comment]
Yeah I double checked with the VO standard and as @juliantaylor it should really only use a dot when outputting products...

**Comment by embray** [comment]
I've also gone through the history of the `VOUnit` format class, and there's no indication that the choice of outputting a space is in any way deliberate.  Very unfortunate this hasn't been caught before.  This is a bug, of course, but it might be worth bringing up on the mailing list too since fixing it may cause changes to some how data is output.

**Comment by eteq** [comment]
Good point, @embray - agreed that it should go out to list for comment before anything gets merged.

**Comment by mhvk** [comment]
Saw this issue while looking through pending ones on units: I definitely agree the output should be with `.`, and @eteq's branch looks like a good implementation. The string roundtripping is only meant to test that strings that conform to the standard round-trip, so the tests can simply be adjusted. As for problems with code already written, I'd hope that mostly it has been used for reading, not writing...

**Comment by pllim** [comment]
Is the 1.0.10 milestone still relevant, given that 1.2 will be out soon?

**Comment by bsipocz** [comment]
@pllim - As I understand LTS, it's supposed to be supported until v2.0 is out sometimes next year?

**Comment by pllim** [comment]
Okay. That's fine. :smile:

**Comment by astrofrog** [comment]
LTS will be supported for two years from its release, so still for a little while. The next LTS release won't necessarily be v2.0 though, it could be v1.5 :)

**Comment by bsipocz** [comment]
@astrofrog - Isn't that confusing? (It definitely confuses my fuzzy logic that v{x}.0 means LTS :) )

**Comment by pllim** [comment]
Is the release numbering logic documented somewhere? I can never keep track of what is what...

**Comment by astrofrog** [comment]
@pllim - we just make it up as we go along. Ha, just kidding, we have an APE:

https://github.com/astropy/astropy-APEs/blob/master/APE2.rst

2.0 is the next LTS, sorry!

**Comment by bsipocz** [comment]
@pllim - It might have been described in e-mails and talks as well, but I guess this is the main written form https://github.com/astropy/astropy-APEs/blob/master/APE2.rst
also: https://github.com/astropy/astropy/wiki/Release-Calendar

**Comment by pllim** [comment]
Thanks! Hard to catch a released APE (har har har).

**Comment by mhvk** [comment]
Still an issue.

**Comment by pllim** [comment]
Perhaps @tomdonaldson can advise, though I think VO Unit spec is still 1.0 as of 2021-04-21? https://www.ivoa.net/documents/VOUnits/index.html
